### DISCUSSION TITLE
Drawn outline on surface possible?

### DISCUSSION THREAD

**Discussion by Psi25** [bnpr/Malt#290] [Q&A] [answered]
Is it possible to draw an "outline" on a surface especially organic surfaces and have it working like an outline?

**Comment by pragma37** [comment, marked as answer]
Do you mean like this?
![imagen](https://user-images.githubusercontent.com/36610999/166934621-8431ed1e-99fa-47d5-b435-df4e6eed3bf1.png)

**Reply by panthavma** [reply]
I would have thought he asked more about being able to draw arbitrary lines and have them treated the same as those detected by the screen space filter.
![image](https://user-images.githubusercontent.com/22663538/166948908-07e4f1be-a422-4dca-aebf-00e738924739.png)

**Reply by pragma37** [reply]
> I would have thought he asked more about being able to draw arbitrary lines and have them treated the same as those detected by the screen space filter.

If that's the case then this is not supported yet. I have freestyle lines working on a private branch, but they still need some work.
The best alternative at the moment is grease pencil.

**Reply by Psi25** [reply]
I meant something like this. I don't know what it's called but I just call it inline(the opposite of outline. is it even a word?).

![outline and line test](https://user-images.githubusercontent.com/103315144/167052954-65ec01da-d830-400a-830d-a28dd8875788.png)

I've been wondering if it's possible like in a texture so it's baked in the mesh. But there's a problem with using a drawn texture line, it's "line" doesn't have a fixed thickness since it's in 3D space. In the example image, I used a grease pencil with shrinkwrap mod, set its stroke thickness to 2D space so its thickness is always constant even when zoomed in.  Grease pencil kind of does the work, but it doesn't work well for mesh deformations.

**Reply by pragma37** [reply]
Sadly, there's nothing built-in to let you achieve this. There are some ways, but you would have to roll your own solution.
I think the easiest approach would be to generate a *signed distance field* texture from a line texture and adjust the threshold based on the pixel_world_size node.

**Reply by Psi25** [reply]
I see. Thanks for the reply.
